modules/apps/depot/depot-service/src/main/java/com/liferay/depot/internal/change/tracking/spi/reference/DepotEntryTableReferenceDefinition.java

```diff
@@ -0,0 +1,50 @@
+/**
+ * SPDX-FileCopyrightText: (c) 2024 Liferay, Inc. https://liferay.com
+ * SPDX-License-Identifier: LGPL-2.1-or-later OR LicenseRef-Liferay-DXP-EULA-2.0.0-2023-06
+ */
+
+package com.liferay.depot.internal.change.tracking.spi.reference;
+
+import com.liferay.change.tracking.spi.reference.TableReferenceDefinition;
+import com.liferay.change.tracking.spi.reference.builder.ChildTableReferenceInfoBuilder;
+import com.liferay.change.tracking.spi.reference.builder.ParentTableReferenceInfoBuilder;
+import com.liferay.depot.model.DepotEntryTable;
+import com.liferay.depot.service.persistence.DepotEntryPersistence;
+import com.liferay.portal.kernel.service.persistence.BasePersistence;
+
+import org.osgi.service.component.annotations.Component;
+import org.osgi.service.component.annotations.Reference;
+
+/**
+ * @author Brooke Dalton
+ */
+@Component(service = TableReferenceDefinition.class)
+public class DepotEntryTableReferenceDefinition
+	implements TableReferenceDefinition<DepotEntryTable> {
+
+	@Override
+	public void defineChildTableReferences(
+		ChildTableReferenceInfoBuilder<DepotEntryTable>
+			childTableReferenceInfoBuilder) {
```

Comment:
DepotEntry adds resources in its add methods, so we should define a child reference with `resourcePermissionReference`.